cmd/tether/ops_linux.go

```diff
@@ -177,33 +168,146 @@ func (t *operations) SetupFirewall(config *tether.ExecutorConfig) error {
 			ifaceName := iface.Attrs().Name
 			log.Debugf("slot %d -> %s", endpoint.ID, ifaceName)
 
-			established.Interface = ifaceName
-			_ = established.Commit(context.TODO())
+			switch endpoint.Network.TrustLevel {
+			case executor.Open:
+				// Accept all incoming and outgoing traffic
+				for _, chain := range []netfilter.Chain{netfilter.Input, netfilter.Output, netfilter.Forward} {
+					(&netfilter.Rule{
+						Chain:     chain,
+						Target:    netfilter.Accept,
+						Interface: ifaceName,
+					}).Commit(context.TODO())
+				}
+
+			case executor.Closed:
+				// Reject all incoming and outgoing traffic
+				// Since our default policy is to drop traffic, nothing is needed here.
+
+			case executor.Outbound:
+				// Reject all incoming traffic, but allow outgoing
+				setupOutboundFirewall(ifaceName)
 
-			// handle the ports
-			for _, p := range endpoint.Ports {
-				// parse the port maps
-				r, err := portToRule(p)
-				if err != nil {
-					log.Errorf("can't apply port rule (%s): %s", p, err.Error())
-					continue
+			case executor.Peers:
+				// Outbound + all ports open to source addresses in --container-network-ip-range
+				setupOutboundFirewall(ifaceName)
+				sourceAddresses := make([]string, len(endpoint.Network.Pools))
+				for i, v := range endpoint.Network.Pools {
```

Comment:
When `--container-network-firewall net:peers` is specified, but `--container-network-ip-range net:...` is not specified, peers will open all traffic on the network interface.  I believe this should be the expected behavior since we have no way of knowing what ip address containers will receive by DHCP and how to classify an ip address as a peer or not.